code/obj/machinery/shower.dm

```diff
@@ -110,4 +110,99 @@
 		src.use_power(50)
 		return
 
+//prototypist requisition: cool futuristic shower, much less messy
+/obj/machinery/sonic_shower
+	name = "sonic shower head"
+	icon = 'icons/obj/stationobjs.dmi'
+	icon_state = "sonicshower"
+	desc = "Removes muck and grime with the amazing power of sound!"
+	dir = NORTH
+	anchored = ANCHORED
+	deconstruct_flags = DECON_WRENCH | DECON_CROWBAR | DECON_WIRECUTTERS
+
+	var/on = 0 //Are we currently spraying???
+	var/tmp/last_spray = 0
+	var/list/clean_desc = list("uncomfortably clean","scoured to the bone","cleansed and repositioned","like your atoms were scrubbed")
+
+	New()
+		..()
+		AddComponent(/datum/component/mechanics_holder)
+		SEND_SIGNAL(src, COMSIG_MECHCOMP_ADD_INPUT, "toggle", PROC_REF(mechcomp_toggle))
+		src.shower_offset()
+
+	proc/mechcomp_toggle()
+		src.toggle(null)
+
+	was_deconstructed_to_frame(mob/user)
+		if (src.on)
+			src.toggle(user)
+
+	was_built_from_frame(mob/user, newly_built)
+		. = ..()
+		src.shower_offset()
+
+	proc/shower_offset()
+		switch(src.dir)
+			if(NORTH)
+				src.pixel_x = 0
+				src.pixel_y = 22
+			if(EAST)
+				src.pixel_x = 8
+				src.pixel_y = 8
+			if(WEST)
+				src.pixel_x = -8
+				src.pixel_y = 8
+			if(SOUTH)
+				src.pixel_x = 0
+				src.pixel_y = 0
+
+	attack_ai(mob/user as mob)
+		src.toggle(user)
+
+	attack_hand(mob/user)
+		src.toggle(user)
+
+	proc/toggle(mob/user)
+		src.on = !src.on
+		if (src.on)
+			SubscribeToProcess()
+		else
+			UnsubscribeProcess()
+		if (user)
+			boutput(user, "You turn [src.on ? "on" : "off"] \the [src].")
+
+	process()
+		if(!on || (world.time < src.last_spray + SPRAY_DELAY))
+			return
+
+		if(status & (NOPOWER))
+			src.on = 0
+			UnsubscribeProcess()
+			return
+
+		src.spray()
+
+	proc/spray()
+		var/cleaned_a_nerd = FALSE
+		src.last_spray = world.time
+
+		for (var/atom/A in range(0, get_turf(src)))
+			if (A == src) continue
+			if (ismob(A))
+				var/mob/M = A
+				if (!isdead(M))
+					M.clean_forensic()
+					M.delStatus("marker_painted")
```

Comment:
should probably affect hygiene motive

Reply:
Did the thing (thought clean_forensic had that built in, for some reason).